browser-extension/mv2/src/client/js/requestResponseRuleHandler.js

```diff
@@ -3,6 +3,19 @@ RQ.RequestResponseRuleHandler.isInitialized = false;
 RQ.RequestResponseRuleHandler.cachedResponseRuleIds = new Set();
 
 RQ.RequestResponseRuleHandler.setup = () => {
+  // TODO: store the extension status in background and watch for changes here
```

Comment:
Extension status is already cached/available in memory to background but I don't think we should watch for changes here. Instead content script should ask background whenever it needs to know the extension status on demand but I guess you are doing this to avoid an async operation here